ilastik/workflows/objectClassification/objectClassificationWorkflow.py

```diff
@@ -670,38 +697,28 @@ class ObjectClassificationWorkflowBinary(ObjectClassificationWorkflow):
     workflowName = "Object Classification (from binary image)"
     workflowDisplayName = "Object Classification [Inputs: Raw Data, Segmentation]"
 
-    def setupInputs(self):
-        data_instructions = 'Use the "Raw Data" tab to load your intensity image(s).\n\n'\
-                            'Use the "Segmentation Image" tab to load your binary mask image(s).'
+    @property
+    def InputImageRoles(self):
+        class ExtraInputImageRoles(SlotNameEnum):
+            SEGMENTATION_IMAGE = super(self.__class__, self).InputImageRoles.getNext()
```

Comment:
I still don't like this part of the code... extending enums seems weird, but it is a bit more readable and safer than the other alternatives I've tried. I'm open to suggestions, though.

We can't have simply `super()` there (I've tried =P) because the `super` word is inside the `ExtraInputImageRoles` class, which means it would try to resolve to a parent of `ExtraInputImageRoles` instead of a parent of the `*Workflow` class that is overriding the `InputImageRoles` property.

Reply:
Why can't we redefine it in each subclass?
```
class InputImageRoles(Enum):
         RAW_DATA  = 0 
         SEGMENTATION_IMAGE = 1
```
Current implementation obcures actual indicies used.

Reply:
I do have to agree, now that I see your comment @m-novikov. While it sort of breaks the "inheritance" here, that @Tomaz-Vieira has nicely established, it is more readable. What do you say, @Tomaz-Vieira

Reply:
The idea was to allow for new data sources to be added to the base `ObjectClassificationWorkflow` and have those be automatically 'inherited' by its children, while also maintaining the order of the UI tabs that represent the data inputs. If we wanted to, say, add a "MASK" input to limit processing to areas within the mask or an "ATLAS" input to map the discovered objects to some some areas of the atlas, then those features would gracefully propagate down the class hierarchy while keeping the child-specific tabs ordered after the base tabs. The "inheritance" style also might help when creating new children of the base workflow class.

I don't mind obscuring the actual indices unless the fact that the indices are sequential also feels obscure; the enum is there to abstract the indices away, anyway. I do agree that it's not very readable, though. If you guys think this will be harder to maintain, then I can switch to something more straightforward.

Reply:
Good that you mentioned. I haven't thought about the serialization, but I just checked and as long as the names are kept the same, the (de)serialization should work fine, even as we add new stuff.